sdk/core/azure-core/src/main/java/com/azure/core/util/ClientOptions.java

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.http.policy.UserAgentPolicy;
+
+import java.util.Collections;
+
+/**
+ * This class represents various options to be set on the client.
+ * <p>
+ * The {@link Header} could be set using {@link ClientOptions#setHeaders(Iterable) setHeaders}. The {@link Header} will
+ * be applied on the request being sent to Azure Service.
+ * <p>
+ * The {@code applicationId} could be set using {@link ClientOptions#setApplicationId(String) setApplicationId} which
+ * is used for setting {@code applicationId} in the {@link UserAgentPolicy}.
+ */
```

Comment:
I'd really like another pass on this JavaDoc, because it is not as descriptive as it would ideally be. It doesn't explain what it means to set various options on the client (and why that is different from other builder parameters), it doesn't explain where users can use this API. It doesn't explain very clearly what a Header is, and how an http or other header might be created and passed in. It doesn't explain what exactly an application ID is and why it might be useful to set it. 

In other words, put yourself in the seat of someone new to this API and be sure you're explaining to them everything they need to know. Same goes for the clientOptions API on builders, especially as the rest of the team will copy / paste that content.